### DISCUSSION TITLE
How do I adding vim global to sumneko_lua for my nvim config files?

### DISCUSSION THREAD

**Discussion by Fyrecean** [VonHeikemen/lsp-zero.nvim#118] [Q&A] [answered]
I'm trying to get rid of this lua warning on every vim config line I make:
``Lua Diagnostics.: Undefined global `vim`. [undefined-global]``

Here is my lsp-zero config file with what I thought would clear the warning:

```
local lsp = require('lsp-zero')

lsp.preset('recommended')

lsp.ensure_installed({
	'sumneko_lua',
	'rust_analyzer',
})

-- This is what I tried to add to remove the warning, but it didn't work
lsp.configure('sumneko_lua', {
    settings = {
        Lua = {
            diagnostics = {
                globals = { 'vim' }
            }
        }
    }
})

lsp.setup({
	enabled = true,
	preselect = cmp.PreselectMode.None
})
```

Any help is appreciated, particularly how I could troubleshoot these types of issues in the future since I'm just starting to learn vim.

**Comment by VonHeikemen** [comment, marked as answer]
That's actually correct.

You might want to check the settings of the "active client":

* First you need to get the `id` of the client. Use this command.

```vim
:LspInfo
```

* Once you have an `id` number use this command

```lua
:lua = vim.lsp.get_client_by_id(1).config.settings
```

> In this example `1` is the `id`.

If your settings are not present then there is another configuration of `sumneko_lua` somewhere else that's overriding this.

---

Side note: the `lsp.setup()` function doesn't take any arguments. And the variable `cmp` might actually be causing an error in the entire file.

**Reply by Fyrecean** [reply]
Thanks!
I ran the command you suggested and the setting was present before I made any changes:
```
settings = {
        Lua = {
          telemetry = {
            enable = false
          }
        },
        diagnostics = {
          globals = { "vim" }
        }
      }
```
I went on to your side note and got it working, looks like I confused a line intended for `cmp.setup` with `lsp.setup` from looking at this [reddit post](https://www.reddit.com/r/neovim/comments/xc555u/how_to_disable_autoselect_for_lsp_suggestions/) which upon further examination, seems dubious anyway...

tl;dr - Removing the argument from `lsp.setup()` solved it

**Reply by VonHeikemen** [reply]
Now it's a good time to mention there is a function called [.nvim_workspace()](https://github.com/VonHeikemen/lsp-zero.nvim#nvim_workspaceopts), it configures `sumneko_lua` for you (so you can delete the `.configure()` call) and adds better autocomplete for neovim's lua api.
